pkg/plugin/plugins.go

```diff
@@ -87,23 +86,36 @@ func (c *PluginsProcessor) Load() error {
 		if yaml.Matcher(f.Name()) {
 			plugins, err := utils.DecodeCRDFromFile(filepath.Join(path, f.Name()), common.Plugin)
 			if err != nil {
-				return fmt.Errorf("failed to load plugin: %v", err)
-			}
-			var plugs []v1.Plugin
-			for _, p := range plugins.([]v1.Plugin) {
-				for _, cp := range c.Plugins {
-					if !isSamePluginSpec(p, cp) {
-						plugs = append(plugs, p)
-					}
-				}
+				return fmt.Errorf("failed to load plugin %v", err)
 			}
-			c.Plugins = append(c.Plugins, plugs...)
+
+			c.Plugins = append(c.Plugins, plugins.([]v1.Plugin)...)
+			i := deduplication(c.Plugins)
+			c.Plugins = i
 		}
 	}
 
 	return nil
 }
 
+func deduplication(plugins []v1.Plugin) []v1.Plugin {
```

Comment:
so what has changed

Reply:
The previous function did not do all the deduplication